src/workerd/api/url.c++

```diff
@@ -17,26 +17,38 @@ namespace {
 
 // Helper functions for the origin, pathname, and search getters and setters.
 
-bool isSpecialScheme(kj::StringPtr scheme) {
-  // TODO(cleanup): Move this to kj::Url.
-  static const std::set<kj::StringPtr> specialSchemes{
-    "ftp", "file", "gopher", "http", "https", "ws", "wss"};
-  return specialSchemes.count(scheme);
-}
-
-kj::Maybe<kj::StringPtr> defaultPortForScheme(kj::StringPtr scheme) {
-  static const std::map<kj::StringPtr, kj::StringPtr> defaultPorts{
-    {"ftp", "21"},
-    {"gopher", "70"},
-    {"http", "80"},
-    {"https", "443"},
-    {"ws", "80"},
-    {"wss", "443"},
-  };
-  auto port = defaultPorts.find(scheme);
-  if (port != defaultPorts.end()) {
-    return port->second;
+constexpr kj::StringPtr is_special_list[] = {
+  "http"_kj, " "_kj, "https"_kj, "ws"_kj, "ftp"_kj, "wss"_kj, "file"_kj, " "_kj};
+constexpr kj::StringPtr special_ports[] = {
+  "80"_kj, ""_kj, "443"_kj, "80"_kj, "21"_kj, "443"_kj, ""_kj, ""_kj};
```

Comment:
nit: a comment here explaining to keep these two lists in sync would be helpful.